FAILED: ClusterAssociationTests.test_association_sameAddressNodeJoin_shouldOverrideExistingNode

https://ci.swiftserver.group/job/swift-distributed-actors-swift51-p1rb/330/console

```
00:14:42.003 Test Case 'ClusterAssociationTests.test_association_sameAddressNodeJoin_shouldOverrideExistingNode' started at 2019-12-23 13:04:27.530
00:14:42.131 2019-12-23T13:04:27+0000 error: [sact://second-REPLACEMENT@localhost:9002][Refs.swift:507][thread:140025228408576][/system] Escalated failure from [/system/cluster] reached top-level guardian [/system], shutting down ActorSystem! Failure was: error(bind(descriptor:ptr:bytes:) failed: Address already in use (errno: 98))
00:14:47.217 Escalated failure from [/system/cluster] reached top-level guardian [/system], shutting down ActorSystem! Failure was: error(bind(descriptor:ptr:bytes:) failed: Address already in use (errno: 98))
00:14:47.217 /code/Tests/DistributedActorsTests/Cluster/AssociationClusteredTests.swift:65: error: ClusterAssociationTests.test_association_sameAddressNodeJoin_shouldOverrideExistingNode : failed - 
00:14:47.217         try shouldNotThrow {
00:14:47.217             ^~~~~~~~~~~~~~~
00:14:47.217 error: Unexpected throw captured
00:14:47.217             try assertAssociated(first, withExactly: [secondReplacement.cluster.node])
00:14:47.217                      ^~~~~~~~~~~
00:14:47.217 error: No result within 5s for block at /code/Tests/DistributedActorsTests/Cluster/AssociationClusteredTests.swift:96. Queried 50 times, within 5s. Last error: TestError(message: "[ActorSystem(first, sact://first@localhost:9001)] did not associate the expected nodes: [[sact://second-REPLACEMENT:1621869467@localhost:9002]]. Associated nodes: [Set([sact://second:3321030862@localhost:9002])], expected nodes: [[sact://second-REPLACEMENT:1621869467@localhost:9002]], diff: [Set([sact://second-REPLACEMENT:1621869467@localhost:9002, sact://second:3321030862@localhost:9002])].")
00:14:47.217 <EXPR>:0: error: ClusterAssociationTests.test_association_sameAddressNodeJoin_shouldOverrideExistingNode : threw error "unexpectedErrorWasThrown"
00:14:47.223 ------------------------------------- ActorSystem(first, sact://first@localhost:9001) ------------------------------------------------
00:14:47.223 Captured log [first][Timers.swift:162]: [trace] Started timer [TimerKey(receptionist/sync)] with generation [0]
00:14:47.223 Captured log [first][ActorShell+Children.swift:346]: [debug] Spawning [DistributedActors.Behavior<DistributedActors.SWIM.Message>.setup((Function))], on path: [/system/cluster/swim]
00:14:47.223 Captured log [first][ActorShell+Children.swift:346]: [debug] Spawning [DistributedActors.Behavior<DistributedActors.ClusterEvent>.setup((Function))], on path: [/system/cluster/leadership]
00:14:47.223 Captured log [first][ClusterShell.swift:253]: [info] Binding to: [sact://first@localhost:9001]
00:14:47.223 Captured log [first][ClusterEventStream.swift:41]: [trace] Successfully subscribed [ActorRef<ClusterEvent>(/system/replicator/$sub-DistributedActors.ClusterEvent-y)], offering membership snapshot
00:14:47.223 Captured log [first][ClusterEventStream.swift:41]: [trace] Successfully subscribed [ActorRef<ClusterEvent>(/system/nodeDeathWatcher/$sub-DistributedActors.ClusterEvent-y)], offering membership snapshot
00:14:47.223 Captured log [first][Timers.swift:162]: [trace] Started timer [TimerKey(swim/periodic-ping)] with generation [0]
00:14:47.223 Captured log [first][Leadership.swift:124]: [trace] Spawned /system/cluster/leadership to run LowestReachableMember(minimumNumberOfMembersToDecide: 2)
00:14:47.223 Captured log [first][Leadership.swift:225]: [trace] Selecting leader among: Membership([Member(sact://first:3249236617@localhost:9001, status: joining, reachability: reachable)])
00:14:47.223 Captured log [first][Leadership.swift:232]: [trace] Not enough members to select leader from, minimum nr of members [1/2]
00:14:47.223 Captured log [first][CRDTReplicatorShell.swift:103]: [trace] Ignoring cluster event snapshot(Membership(Dictionary.Values([]))
00:14:47.223 \- metadata: "crdt/replicator": /system/replicator, "crdt/replicator/remoteReplicators": []
00:14:47.223 Captured log [first][ClusterEventStream.swift:41]: [trace] Successfully subscribed [ActorRef<ClusterEvent>(/system/cluster/leadership)], offering membership snapshot
00:14:47.223 Captured log [first][ClusterShell.swift:264]: [info] Bound to [IPv4]127.0.0.1/127.0.0.1:9001
00:14:47.223 Captured log [first][ActorShell+Children.swift:346]: [debug] Spawning [DistributedActors.Behavior<DistributedActors.PeriodicBroadcastShell<DistributedActors.ClusterShell.Message>.Message>.setup((Function))], on path: [/system/cluster/periodicBroadcast]
00:14:47.223 Captured log [first][Timers.swift:162]: [trace] Started timer [TimerKey(tick)] with generation [0]
00:14:47.223 Captured log [first][ClusterShellState.swift:381]: [trace] Applied change via cluster event: sact://first@localhost:9001 :: [unknown] -> [joining]
00:14:47.223 \- metadata: "membership/count": 1
00:14:47.223 Captured log [first][ClusterShellState.swift:407]: [trace] Membership updated Membership sact://first@localhost:9001:
00:14:47.223    LEADER: [.none]
00:14:47.223    [sact://first:3249236617@localhost:9001] STATUS: [joining], by membershipChange(sact://first@localhost:9001 :: [joining] -> [joining])
00:14:47.223 Captured log [first][ClusterEventStream.swift:58]: [trace] Published event membershipChange(sact://first@localhost:9001 :: [joining] -> [joining]) to 3 subscribers
00:14:47.223 Captured log [first][CRDTReplicatorShell.swift:103]: [trace] Ignoring cluster event membershipChange(sact://first@localhost:9001 :: [joining] -> [joining])
00:14:47.223 \- metadata: "crdt/replicator": /system/replicator, "crdt/replicator/remoteReplicators": []
00:14:47.223 Captured log [first][Leadership.swift:225]: [trace] Selecting leader among: Membership([Member(sact://first:3249236617@localhost:9001, status: joining, reachability: reachable)])
00:14:47.223 Captured log [first][Leadership.swift:232]: [trace] Not enough members to select leader from, minimum nr of members [1/2]
00:14:47.223 Captured log [first][ClusterEventStream.swift:41]: [trace] Successfully subscribed [ActorRef<ClusterEvent>(/user/$testProbe-0/$messageAdapter)], offering membership snapshot
00:14:47.223 Captured log [first][ClusterShell.swift:476]: [info] Extending handshake offer to sact://second@localhost:9002)
00:14:47.223 Captured log [first][ClusterShell.swift:632]: [debug] Associated with: [sact://second:3321030862@localhost:9002]; Membership change: sact://second@localhost:9002 :: [unknown] -> [joining], resulting in: Membership([Member(sact://first:3249236617@localhost:9001, status: joining, reachability: reachable), Member(sact://second:3321030862@localhost:9002, status: joining, reachability: reachable)])
00:14:47.223 Captured log [first][ClusterEventStream.swift:58]: [trace] Published event membershipChange(sact://second@localhost:9002 :: [unknown] -> [joining]) to 4 subscribers
00:14:47.223 Captured log [first][Leadership.swift:225]: [trace] Selecting leader among: Membership([Member(sact://second:3321030862@localhost:9002, status: joining, reachability: reachable), Member(sact://first:3249236617@localhost:9001, status: joining, reachability: reachable)])
00:14:47.223 Captured log [first][Leadership.swift:251]: [trace] Selected leader: [[Optional(Member(sact://first:3249236617@localhost:9001, status: joining, reachability: reachable))]], out of Membership([Member(sact://second:3321030862@localhost:9002, status: joining, reachability: reachable), Member(sact://first:3249236617@localhost:9001, status: joining, reachability: reachable)])
00:14:47.223 Captured log [first][CRDTReplicatorShell.swift:103]: [trace] Ignoring cluster event membershipChange(sact://second@localhost:9002 :: [unknown] -> [joining])
00:14:47.223 \- metadata: "crdt/replicator": /system/replicator, "crdt/replicator/remoteReplicators": []
00:14:47.223 Captured log [first][ClusterShellState.swift:370]: [info] Leader change: LeadershipChange(oldLeader: nil, newLeader: Optional(Member(sact://first:3249236617@localhost:9001, status: joining, reachability: reachable)))
00:14:47.223 \- metadata: "membership/count": 2
00:14:47.223 Captured log [first][ClusterShellState.swift:447]: [info] Leader moving member: sact://first@localhost:9001 :: [joining] -> [     up]
00:14:47.223 Captured log [first][ClusterShellState.swift:447]: [info] Leader moving member: sact://second@localhost:9002 :: [joining] -> [     up]
00:14:47.223 Captured log [first][ClusterShellState.swift:407]: [trace] Membership updated Membership sact://first@localhost:9001:
00:14:47.223    LEADER: [Member(sact://first@localhost:9001, status: joining, reachability: reachable)]
00:14:47.223    [sact://first:3249236617@localhost:9001] STATUS: [     up]
00:14:47.223    [sact://second:3321030862@localhost:9002] STATUS: [     up], by leadershipChange(DistributedActors.LeadershipChange(oldLeader: nil, newLeader: Optional(Member(sact://first:3249236617@localhost:9001, status: joining, reachability: reachable))))
00:14:47.223 Captured log [first][ClusterEventStream.swift:58]: [trace] Published event membershipChange(sact://first@localhost:9001 :: [joining] -> [     up]) to 4 subscribers
00:14:47.223 Captured log [first][ClusterEventStream.swift:58]: [trace] Published event membershipChange(sact://second@localhost:9002 :: [joining] -> [     up]) to 4 subscribers
00:14:47.223 Captured log [first][ClusterEventStream.swift:58]: [trace] Published event leadershipChange(DistributedActors.LeadershipChange(oldLeader: nil, newLeader: Optional(Member(sact://first:3249236617@localhost:9001, status: joining, reachability: reachable)))) to 4 subscribers
00:14:47.223 Captured log [first][Leadership.swift:225]: [trace] Selecting leader among: Membership([Member(sact://second:3321030862@localhost:9002, status: joining, reachability: reachable), Member(sact://first:3249236617@localhost:9001, status: up, reachability: reachable)])
00:14:47.223 Captured log [first][Leadership.swift:251]: [trace] Selected leader: [[Optional(Member(sact://first:3249236617@localhost:9001, status: up, reachability: reachable))]], out of Membership([Member(sact://second:3321030862@localhost:9002, status: joining, reachability: reachable), Member(sact://first:3249236617@localhost:9001, status: up, reachability: reachable)])
00:14:47.223 Captured log [first][CRDTReplicatorShell.swift:93]: [trace] Skip adding member Member(sact://first@localhost:9001, status: up, reachability: reachable) to replicator because it is the same as local node
00:14:47.223 \- metadata: "crdt/replicator": /system/replicator, "crdt/replicator/remoteReplicators": []
00:14:47.223 Captured log [first][CRDTReplicatorShell.swift:103]: [trace] Ignoring cluster event leadershipChange(DistributedActors.LeadershipChange(oldLeader: nil, newLeader: Optional(Member(sact://first:3249236617@localhost:9001, status: joining, reachability: reachable))))
00:14:47.224 \- metadata: "crdt/replicator": /system/replicator, "crdt/replicator/remoteReplicators": [ActorRef<CRDT.Replicator.Message>(sact://second@localhost:9002/system/replicator)]
00:14:47.224 Captured log [first][ClusterShell.swift:731]: [debug] Associated sact://second@localhost:9002, informing SWIM to monitor this node.
00:14:47.224 Captured log [first][Leadership.swift:225]: [trace] Selecting leader among: Membership([Member(sact://second:3321030862@localhost:9002, status: up, reachability: reachable), Member(sact://first:3249236617@localhost:9001, status: up, reachability: reachable)])
00:14:47.224 Captured log [first][Leadership.swift:251]: [trace] Selected leader: [[Optional(Member(sact://first:3249236617@localhost:9001, status: up, reachability: reachable))]], out of Membership([Member(sact://second:3321030862@localhost:9002, status: up, reachability: reachable), Member(sact://first:3249236617@localhost:9001, status: up, reachability: reachable)])
00:14:47.224 Captured log [first][SWIMShell.swift:145]: [trace] Sending ping to [ActorRef<SWIM.Message>(sact://second@localhost:9002/system/cluster/swim)] with payload [membership([DistributedActors.SWIMMember(ref: ActorRef<SWIM.Message>(/system/cluster/swim), status: DistributedActors.SWIM.Status.alive(incarnation: 0), protocolPeriod: 0)])]
00:14:47.224 Captured log [first][SWIMShell.swift:425]: [trace] Requesting handshake with sact://second@localhost:9002
00:14:47.224 Captured log [first][ClusterShell.swift:430]: [debug] Attempted associating with already associated node: [sact://second@localhost:9002], existing association: [associated(AssociatedState(channel: SocketChannel { BaseSocket { fd=22 }, active = true, localAddress = Optional([IPv4]127.0.0.1/127.0.0.1:44174), remoteAddress = Optional([IPv4]127.0.0.1/127.0.0.1:9002) }, selfNode: sact://first@localhost:9001, remoteNode: sact://second@localhost:9002))]
00:14:47.224 Captured log [first][SWIMShell.swift:145]: [trace] Sending ping to [ActorRef<SWIM.Message>(sact://second@localhost:9002/system/cluster/swim)] with payload [membership([DistributedActors.SWIMMember(ref: ActorRef<SWIM.Message>(sact://second@localhost:9002/system/cluster/swim), status: DistributedActors.SWIM.Status.alive(incarnation: 0), protocolPeriod: 0), DistributedActors.SWIMMember(ref: ActorRef<SWIM.Message>(/system/cluster/swim), status: DistributedActors.SWIM.Status.alive(incarnation: 0), protocolPeriod: 0)])]
00:14:47.224 Captured log [first][SWIMShell.swift:234]: [trace] Received ack from [ActorRef<SWIM.Message>(sact://second@localhost:9002/system/cluster/swim)] with incarnation [0] and payload [membership([DistributedActors.SWIMMember(ref: ActorRef<SWIM.Message>(sact://second@localhost:9002/system/cluster/swim), status: DistributedActors.SWIM.Status.alive(incarnation: 0), protocolPeriod: 0)])]
00:14:47.224 Captured log [first][SWIMShell.swift:366]: [trace] Ignoring gossip about member [sact://second:3321030862@localhost:9002], incoming: [alive(incarnation: 0)] does not supersede current: [alive(incarnation: 0)]
00:14:47.224 Captured log [first][SWIMShell.swift:234]: [trace] Received ack from [ActorRef<SWIM.Message>(sact://second@localhost:9002/system/cluster/swim)] with incarnation [0] and payload [membership([DistributedActors.SWIMMember(ref: ActorRef<SWIM.Message>(/system/cluster/swim), status: DistributedActors.SWIM.Status.alive(incarnation: 0), protocolPeriod: 0), DistributedActors.SWIMMember(ref: ActorRef<SWIM.Message>(sact://second@localhost:9002/system/cluster/swim), status: DistributedActors.SWIM.Status.alive(incarnation: 0), protocolPeriod: 0)])]
00:14:47.224 Captured log [first][SWIMShell.swift:366]: [trace] Ignoring gossip about member [sact://second:3321030862@localhost:9002], incoming: [alive(incarnation: 0)] does not supersede current: [alive(incarnation: 0)]
00:14:47.224 Captured log [first][Timers.swift:117]: [trace] Cancel timer [TimerKey(tick)] with generation [0]
00:14:47.224 Captured log [first][SWIMShell.swift:259]: [trace] Received periodic trigger to ping random member, among: 2
00:14:47.224 \- metadata: "swim/incarnation": 0, "swim/memberCount": 2, "swim/protocolPeriod": 0, "swim/suspectCount": 0
00:14:47.224 Captured log [first][SWIMShell.swift:145]: [trace] Sending ping to [ActorRef<SWIM.Message>(sact://second@localhost:9002/system/cluster/swim)] with payload [membership([DistributedActors.SWIMMember(ref: ActorRef<SWIM.Message>(sact://second@localhost:9002/system/cluster/swim), status: DistributedActors.SWIM.Status.alive(incarnation: 0), protocolPeriod: 0), DistributedActors.SWIMMember(ref: ActorRef<SWIM.Message>(/system/cluster/swim), status: DistributedActors.SWIM.Status.alive(incarnation: 0), protocolPeriod: 0)])]
00:14:47.224 Captured log [first][Timers.swift:162]: [trace] Started timer [TimerKey(tick)] with generation [1]
00:14:47.224 Captured log [first][SWIMShell.swift:224]: [warning] Did not receive ack from [sact://second:3321030862@localhost:9002/system/cluster/swim] within [300ms]. Sending ping requests to other members.
00:14:47.224 Captured log [first][SWIMShell.swift:174]: [info] No members to ping-req through, marking [ActorRef<SWIM.Message>(sact://second@localhost:9002/system/cluster/swim)] immediately as [.suspect]. Members: [[ActorRef<SWIM.Message>(/system/cluster/swim): DistributedActors.SWIM.Status.alive(incarnation: 0), ActorRef<SWIM.Message>(sact://second@localhost:9002/system/cluster/swim): DistributedActors.SWIM.Status.alive(incarnation: 0)]]
00:14:47.224 Captured log [first][SWIMShell.swift:259]: [trace] Received periodic trigger to ping random member, among: 2
00:14:47.224 \- metadata: "swim/incarnation": 0, "swim/memberCount": 2, "swim/protocolPeriod": 1, "swim/suspectCount": 1
00:14:47.224 Captured log [first][SWIMShell.swift:145]: [trace] Sending ping to [ActorRef<SWIM.Message>(sact://second@localhost:9002/system/cluster/swim)] with payload [membership([DistributedActors.SWIMMember(ref: ActorRef<SWIM.Message>(sact://second@localhost:9002/system/cluster/swim), status: DistributedActors.SWIM.Status.suspect(incarnation: 0), protocolPeriod: 1), DistributedActors.SWIMMember(ref: ActorRef<SWIM.Message>(/system/cluster/swim), status: DistributedActors.SWIM.Status.alive(incarnation: 0), protocolPeriod: 0)])]
00:14:47.224 Captured log [first][Timers.swift:117]: [trace] Cancel timer [TimerKey(tick)] with generation [1]
00:14:47.224 Captured log [first][Timers.swift:162]: [trace] Started timer [TimerKey(tick)] with generation [2]
00:14:47.224 Captured log [first][SWIMShell.swift:224]: [warning] Did not receive ack from [sact://second:3321030862@localhost:9002/system/cluster/swim] within [300ms]. Sending ping requests to other members.
00:14:47.224 Captured log [first][SWIMShell.swift:174]: [info] No members to ping-req through, marking [ActorRef<SWIM.Message>(sact://second@localhost:9002/system/cluster/swim)] immediately as [.suspect]. Members: [[ActorRef<SWIM.Message>(/system/cluster/swim): DistributedActors.SWIM.Status.alive(incarnation: 0), ActorRef<SWIM.Message>(sact://second@localhost:9002/system/cluster/swim): DistributedActors.SWIM.Status.suspect(incarnation: 0)]]
00:14:47.224 Captured log [first][SWIMShell.swift:259]: [trace] Received periodic trigger to ping random member, among: 2
00:14:47.224 \- metadata: "swim/incarnation": 0, "swim/memberCount": 2, "swim/protocolPeriod": 2, "swim/suspectCount": 1
00:14:47.224 Captured log [first][SWIMShell.swift:145]: [trace] Sending ping to [ActorRef<SWIM.Message>(sact://second@localhost:9002/system/cluster/swim)] with payload [membership([DistributedActors.SWIMMember(ref: ActorRef<SWIM.Message>(sact://second@localhost:9002/system/cluster/swim), status: DistributedActors.SWIM.Status.suspect(incarnation: 0), protocolPeriod: 1), DistributedActors.SWIMMember(ref: ActorRef<SWIM.Message>(/system/cluster/swim), status: DistributedActors.SWIM.Status.alive(incarnation: 0), protocolPeriod: 0)])]
00:14:47.224 Captured log [first][Timers.swift:117]: [trace] Cancel timer [TimerKey(tick)] with generation [2]
00:14:47.224 Captured log [first][Timers.swift:162]: [trace] Started timer [TimerKey(tick)] with generation [3]
00:14:47.224 Captured log [first][SWIMShell.swift:224]: [warning] Did not receive ack from [sact://second:3321030862@localhost:9002/system/cluster/swim] within [300ms]. Sending ping requests to other members.
00:14:47.224 Captured log [first][SWIMShell.swift:174]: [info] No members to ping-req through, marking [ActorRef<SWIM.Message>(sact://second@localhost:9002/system/cluster/swim)] immediately as [.suspect]. Members: [[ActorRef<SWIM.Message>(/system/cluster/swim): DistributedActors.SWIM.Status.alive(incarnation: 0), ActorRef<SWIM.Message>(sact://second@localhost:9002/system/cluster/swim): DistributedActors.SWIM.Status.suspect(incarnation: 0)]]
00:14:47.224 Captured log [first][SWIMShell.swift:259]: [trace] Received periodic trigger to ping random member, among: 2
00:14:47.224 \- metadata: "swim/incarnation": 0, "swim/memberCount": 2, "swim/protocolPeriod": 3, "swim/suspectCount": 1
00:14:47.224 Captured log [first][SWIMShell.swift:145]: [trace] Sending ping to [ActorRef<SWIM.Message>(sact://second@localhost:9002/system/cluster/swim)] with payload [membership([DistributedActors.SWIMMember(ref: ActorRef<SWIM.Message>(sact://second@localhost:9002/system/cluster/swim), status: DistributedActors.SWIM.Status.suspect(incarnation: 0), protocolPeriod: 1)])]
00:14:47.224 Captured log [first][Timers.swift:117]: [trace] Cancel timer [TimerKey(tick)] with generation [3]
00:14:47.224 Captured log [first][Timers.swift:162]: [trace] Started timer [TimerKey(tick)] with generation [4]
00:14:47.224 Captured log [first][SWIMShell.swift:224]: [warning] Did not receive ack from [sact://second:3321030862@localhost:9002/system/cluster/swim] within [300ms]. Sending ping requests to other members.
00:14:47.224 Captured log [first][SWIMShell.swift:174]: [info] No members to ping-req through, marking [ActorRef<SWIM.Message>(sact://second@localhost:9002/system/cluster/swim)] immediately as [.suspect]. Members: [[ActorRef<SWIM.Message>(/system/cluster/swim): DistributedActors.SWIM.Status.alive(incarnation: 0), ActorRef<SWIM.Message>(sact://second@localhost:9002/system/cluster/swim): DistributedActors.SWIM.Status.suspect(incarnation: 0)]]
00:14:47.224 Captured log [first][SWIMShell.swift:259]: [trace] Received periodic trigger to ping random member, among: 2
00:14:47.224 \- metadata: "swim/incarnation": 0, "swim/memberCount": 2, "swim/protocolPeriod": 4, "swim/suspectCount": 1
00:14:47.224 Captured log [first][SWIMShell.swift:145]: [trace] Sending ping to [ActorRef<SWIM.Message>(sact://second@localhost:9002/system/cluster/swim)] with payload [membership([DistributedActors.SWIMMember(ref: ActorRef<SWIM.Message>(sact://second@localhost:9002/system/cluster/swim), status: DistributedActors.SWIM.Status.suspect(incarnation: 0), protocolPeriod: 1)])]
00:14:47.224 Captured log [first][Timers.swift:117]: [trace] Cancel timer [TimerKey(tick)] with generation [4]
00:14:47.224 Captured log [first][Timers.swift:162]: [trace] Started timer [TimerKey(tick)] with generation [5]
00:14:47.224 ========================================================================================================================
00:14:47.224 ------------------------------------- ActorSystem(second, sact://second@localhost:9002) ------------------------------------------------
00:14:47.224 Captured log [second][ClusterEventStream.swift:41]: [trace] Successfully subscribed [ActorRef<ClusterEvent>(/user/$testProbe-0/$messageAdapter)], offering membership snapshot
00:14:47.224 Captured log [second][Timers.swift:162]: [trace] Started timer [TimerKey(receptionist/sync)] with generation [0]
00:14:47.224 Captured log [second][ActorShell+Children.swift:346]: [debug] Spawning [DistributedActors.Behavior<DistributedActors.SWIM.Message>.setup((Function))], on path: [/system/cluster/swim]
00:14:47.224 Captured log [second][ActorShell+Children.swift:346]: [debug] Spawning [DistributedActors.Behavior<DistributedActors.ClusterEvent>.setup((Function))], on path: [/system/cluster/leadership]
00:14:47.224 Captured log [second][ClusterShell.swift:253]: [info] Binding to: [sact://second@localhost:9002]
00:14:47.224 Captured log [second][ClusterEventStream.swift:41]: [trace] Successfully subscribed [ActorRef<ClusterEvent>(/system/replicator/$sub-DistributedActors.ClusterEvent-y)], offering membership snapshot
00:14:47.224 Captured log [second][ClusterEventStream.swift:41]: [trace] Successfully subscribed [ActorRef<ClusterEvent>(/system/nodeDeathWatcher/$sub-DistributedActors.ClusterEvent-y)], offering membership snapshot
00:14:47.224 Captured log [second][Timers.swift:162]: [trace] Started timer [TimerKey(swim/periodic-ping)] with generation [0]
00:14:47.224 Captured log [second][Leadership.swift:124]: [trace] Spawned /system/cluster/leadership to run LowestReachableMember(minimumNumberOfMembersToDecide: 2)
00:14:47.224 Captured log [second][Leadership.swift:225]: [trace] Selecting leader among: Membership([Member(sact://second:3321030862@localhost:9002, status: joining, reachability: reachable)])
00:14:47.224 Captured log [second][Leadership.swift:232]: [trace] Not enough members to select leader from, minimum nr of members [1/2]
00:14:47.224 Captured log [second][CRDTReplicatorShell.swift:103]: [trace] Ignoring cluster event snapshot(Membership(Dictionary.Values([]))
00:14:47.224 \- metadata: "crdt/replicator": /system/replicator, "crdt/replicator/remoteReplicators": []
00:14:47.224 Captured log [second][ClusterEventStream.swift:41]: [trace] Successfully subscribed [ActorRef<ClusterEvent>(/system/cluster/leadership)], offering membership snapshot
00:14:47.224 Captured log [second][ClusterShell.swift:264]: [info] Bound to [IPv4]127.0.0.1/127.0.0.1:9002
00:14:47.224 Captured log [second][ActorShell+Children.swift:346]: [debug] Spawning [DistributedActors.Behavior<DistributedActors.PeriodicBroadcastShell<DistributedActors.ClusterShell.Message>.Message>.setup((Function))], on path: [/system/cluster/periodicBroadcast]
00:14:47.224 Captured log [second][Timers.swift:162]: [trace] Started timer [TimerKey(tick)] with generation [0]
00:14:47.224 Captured log [second][ClusterShellState.swift:381]: [trace] Applied change via cluster event: sact://second@localhost:9002 :: [unknown] -> [joining]
00:14:47.224 \- metadata: "membership/count": 1
00:14:47.224 Captured log [second][ClusterShellState.swift:407]: [trace] Membership updated Membership sact://second@localhost:9002:
00:14:47.224    LEADER: [.none]
00:14:47.224    [sact://second:3321030862@localhost:9002] STATUS: [joining], by membershipChange(sact://second@localhost:9002 :: [joining] -> [joining])
00:14:47.224 Captured log [second][ClusterEventStream.swift:58]: [trace] Published event membershipChange(sact://second@localhost:9002 :: [joining] -> [joining]) to 4 subscribers
00:14:47.224 Captured log [second][CRDTReplicatorShell.swift:103]: [trace] Ignoring cluster event membershipChange(sact://second@localhost:9002 :: [joining] -> [joining])
00:14:47.224 \- metadata: "crdt/replicator": /system/replicator, "crdt/replicator/remoteReplicators": []
00:14:47.224 Captured log [second][ClusterShell.swift:519]: [info] Accept association with [sact://first:3249236617@localhost:9001]!
00:14:47.224 Captured log [second][Leadership.swift:225]: [trace] Selecting leader among: Membership([Member(sact://second:3321030862@localhost:9002, status: joining, reachability: reachable)])
00:14:47.224 Captured log [second][Leadership.swift:232]: [trace] Not enough members to select leader from, minimum nr of members [1/2]
00:14:47.224 Captured log [second][ClusterEventStream.swift:58]: [trace] Published event membershipChange(sact://first@localhost:9001 :: [unknown] -> [joining]) to 4 subscribers
00:14:47.224 Captured log [second][CRDTReplicatorShell.swift:103]: [trace] Ignoring cluster event membershipChange(sact://first@localhost:9001 :: [unknown] -> [joining])
00:14:47.224 \- metadata: "crdt/replicator": /system/replicator, "crdt/replicator/remoteReplicators": []
00:14:47.224 Captured log [second][Leadership.swift:225]: [trace] Selecting leader among: Membership([Member(sact://second:3321030862@localhost:9002, status: joining, reachability: reachable), Member(sact://first:3249236617@localhost:9001, status: joining, reachability: reachable)])
00:14:47.224 Captured log [second][Leadership.swift:251]: [trace] Selected leader: [[Optional(Member(sact://first:3249236617@localhost:9001, status: joining, reachability: reachable))]], out of Membership([Member(sact://second:3321030862@localhost:9002, status: joining, reachability: reachable), Member(sact://first:3249236617@localhost:9001, status: joining, reachability: reachable)])
00:14:47.224 Captured log [second][ClusterShellState.swift:370]: [info] Leader change: LeadershipChange(oldLeader: nil, newLeader: Optional(Member(sact://first:3249236617@localhost:9001, status: joining, reachability: reachable)))
00:14:47.224 \- metadata: "membership/count": 2
00:14:47.224 Captured log [second][ClusterShellState.swift:407]: [trace] Membership updated Membership sact://second@localhost:9002:
00:14:47.224    LEADER: [Member(sact://first@localhost:9001, status: joining, reachability: reachable)]
00:14:47.224    [sact://first:3249236617@localhost:9001] STATUS: [joining]
00:14:47.224    [sact://second:3321030862@localhost:9002] STATUS: [joining], by leadershipChange(DistributedActors.LeadershipChange(oldLeader: nil, newLeader: Optional(Member(sact://first:3249236617@localhost:9001, status: joining, reachability: reachable))))
00:14:47.228 Captured log [second][ClusterEventStream.swift:58]: [trace] Published event leadershipChange(DistributedActors.LeadershipChange(oldLeader: nil, newLeader: Optional(Member(sact://first:3249236617@localhost:9001, status: joining, reachability: reachable)))) to 4 subscribers
00:14:47.228 Captured log [second][CRDTReplicatorShell.swift:103]: [trace] Ignoring cluster event leadershipChange(DistributedActors.LeadershipChange(oldLeader: nil, newLeader: Optional(Member(sact://first:3249236617@localhost:9001, status: joining, reachability: reachable))))
00:14:47.228 \- metadata: "crdt/replicator": /system/replicator, "crdt/replicator/remoteReplicators": []
00:14:47.228 Captured log [second][SWIMShell.swift:425]: [trace] Requesting handshake with sact://first@localhost:9001
00:14:47.228 Captured log [second][ClusterShell.swift:430]: [debug] Attempted associating with already associated node: [sact://first@localhost:9001], existing association: [associated(AssociatedState(channel: SocketChannel { BaseSocket { fd=23 }, active = true, localAddress = Optional([IPv4]127.0.0.1/127.0.0.1:9002), remoteAddress = Optional([IPv4]127.0.0.1/127.0.0.1:44174) }, selfNode: sact://second@localhost:9002, remoteNode: sact://first@localhost:9001))]
00:14:47.228 Captured log [second][SWIMShell.swift:145]: [trace] Sending ping to [ActorRef<SWIM.Message>(sact://first@localhost:9001/system/cluster/swim)] with payload [membership([DistributedActors.SWIMMember(ref: ActorRef<SWIM.Message>(sact://first@localhost:9001/system/cluster/swim), status: DistributedActors.SWIM.Status.alive(incarnation: 0), protocolPeriod: 0), DistributedActors.SWIMMember(ref: ActorRef<SWIM.Message>(/system/cluster/swim), status: DistributedActors.SWIM.Status.alive(incarnation: 0), protocolPeriod: 0)])]
00:14:47.228 Captured log [second][SWIMShell.swift:366]: [trace] Ignoring gossip about member [sact://first:3249236617@localhost:9001], incoming: [alive(incarnation: 0)] does not supersede current: [alive(incarnation: 0)]
00:14:47.228 Captured log [second][SWIMShell.swift:234]: [trace] Received ack from [ActorRef<SWIM.Message>(sact://first@localhost:9001/system/cluster/swim)] with incarnation [0] and payload [membership([DistributedActors.SWIMMember(ref: ActorRef<SWIM.Message>(sact://first@localhost:9001/system/cluster/swim), status: DistributedActors.SWIM.Status.alive(incarnation: 0), protocolPeriod: 0)])]
00:14:47.228 Captured log [second][SWIMShell.swift:366]: [trace] Ignoring gossip about member [sact://first:3249236617@localhost:9001], incoming: [alive(incarnation: 0)] does not supersede current: [alive(incarnation: 0)]
00:14:47.228 Captured log [second][ClusterEventStream.swift:66]: [trace] Removed subscriber [/user/$testProbe-0/$messageAdapter], because it terminated
00:14:47.228 Captured log [second][Timers.swift:117]: [trace] Cancel timer [TimerKey(receptionist/sync)] with generation [0]
00:14:47.228 Captured log [second][Timers.swift:117]: [trace] Cancel timer [TimerKey(tick)] with generation [0]
00:14:47.228 Captured log [second][Timers.swift:117]: [trace] Cancel timer [TimerKey(swim/periodic-ping)] with generation [0]
00:14:47.228 ========================================================================================================================
00:14:47.228 ------------------------------------- ActorSystem(second-REPLACEMENT, sact://second-REPLACEMENT@localhost:9002) ------------------------------------------------
00:14:47.228 Captured log [second-REPLACEMENT][ClusterEventStream.swift:41]: [trace] Successfully subscribed [ActorRef<ClusterEvent>(/user/$testProbe-0/$messageAdapter)], offering membership snapshot
00:14:47.228 Captured log [second-REPLACEMENT][Timers.swift:162]: [trace] Started timer [TimerKey(receptionist/sync)] with generation [0]
00:14:47.228 Captured log [second-REPLACEMENT][ActorShell+Children.swift:346]: [debug] Spawning [DistributedActors.Behavior<DistributedActors.SWIM.Message>.setup((Function))], on path: [/system/cluster/swim]
00:14:47.228 Captured log [second-REPLACEMENT][ActorShell+Children.swift:346]: [debug] Spawning [DistributedActors.Behavior<DistributedActors.ClusterEvent>.setup((Function))], on path: [/system/cluster/leadership]
00:14:47.237 Captured log [second-REPLACEMENT][ClusterShell.swift:253]: [info] Binding to: [sact://second-REPLACEMENT@localhost:9002]
00:14:47.237 Captured log [second-REPLACEMENT][ClusterEventStream.swift:41]: [trace] Successfully subscribed [ActorRef<ClusterEvent>(/system/replicator/$sub-DistributedActors.ClusterEvent-y)], offering membership snapshot
00:14:47.237 Captured log [second-REPLACEMENT][ClusterEventStream.swift:41]: [trace] Successfully subscribed [ActorRef<ClusterEvent>(/system/nodeDeathWatcher/$sub-DistributedActors.ClusterEvent-y)], offering membership snapshot
00:14:47.237 Captured log [second-REPLACEMENT][Timers.swift:162]: [trace] Started timer [TimerKey(swim/periodic-ping)] with generation [0]
00:14:47.237 Captured log [second-REPLACEMENT][Leadership.swift:124]: [trace] Spawned /system/cluster/leadership to run LowestReachableMember(minimumNumberOfMembersToDecide: 2)
00:14:47.237 Captured log [second-REPLACEMENT][Leadership.swift:225]: [trace] Selecting leader among: Membership([Member(sact://second-REPLACEMENT:1621869467@localhost:9002, status: joining, reachability: reachable)])
00:14:47.237 Captured log [second-REPLACEMENT][Leadership.swift:232]: [trace] Not enough members to select leader from, minimum nr of members [1/2]
00:14:47.237 Captured log [second-REPLACEMENT][CRDTReplicatorShell.swift:103]: [trace] Ignoring cluster event snapshot(Membership(Dictionary.Values([]))
00:14:47.237 \- metadata: "crdt/replicator": /system/replicator, "crdt/replicator/remoteReplicators": []
00:14:47.237 Captured log [second-REPLACEMENT][ClusterEventStream.swift:41]: [trace] Successfully subscribed [ActorRef<ClusterEvent>(/system/cluster/leadership)], offering membership snapshot
00:14:47.237 Captured log [second-REPLACEMENT][Supervision.swift:515]: [warning] Actor has THROWN [bind(descriptor:ptr:bytes:) failed: Address already in use (errno: 98)]:IOError while interpreting continuation((Function)), handling with [.escalate<AllFailures>] supervision strategy
00:14:47.237 Captured log [second-REPLACEMENT][Timers.swift:117]: [trace] Cancel timer [TimerKey(swim/periodic-ping)] with generation [0]
00:14:47.237 Captured log [second-REPLACEMENT][ClusterEventStream.swift:66]: [trace] Removed subscriber [/system/cluster/leadership], because it terminated
00:14:47.237 Captured log [second-REPLACEMENT][ClusterEventStream.swift:66]: [trace] Removed subscriber [/user/$testProbe-0/$messageAdapter], because it terminated
00:14:47.237 Captured log [second-REPLACEMENT][Timers.swift:117]: [trace] Cancel timer [TimerKey(receptionist/sync)] with generation [0]
00:14:47.237 ========================================================================================================================
00:14:47.237 Test Case 'ClusterAssociationTests.test_association_sameAddressNodeJoin_shouldOverrideExistingNode' failed (5.226 seconds)
```

--- a/Tests/DistributedActorsTests/Cluster/AssociationClusteredTests.swift
+++ b/Tests/DistributedActorsTests/Cluster/AssociationClusteredTests.swift
@@ -79,7 +79,8 @@ final class ClusterAssociationTests: ClusteredNodesTestBase {
             try assertAssociated(second, withExactly: first.cluster.node)
 
             let oldSecond = second
-            oldSecond.shutdown() // kill remote node
+            let shutdown = oldSecond.shutdown() // kill remote node
+            try shutdown.wait(atMost: .seconds(3))
 
             let secondReplacement = self.setUpNode(secondName + "-REPLACEMENT") { settings in
                 settings.cluster.bindPort = remotePort

--- a/Tests/DistributedActorsTests/Cluster/ClusteredNodesTestBase.swift
+++ b/Tests/DistributedActorsTests/Cluster/ClusteredNodesTestBase.swift
@@ -221,8 +221,12 @@ extension ClusteredNodesTestBase {
                 var diff = Set(associatedNodes)
                 diff.formSymmetricDifference(exactlyNodes)
                 guard diff.isEmpty else {
-                    throw TestError("[\(system)] did not associate the expected nodes: [\(exactlyNodes)]. " +
-                        "Associated nodes: \(reflecting: associatedNodes), expected nodes: \(reflecting: exactlyNodes), diff: \(reflecting: diff).")
+                    throw TestError(
+                        """
+                        [\(system)] did not associate the expected nodes: [\(exactlyNodes)].
+                          Associated nodes: \(reflecting: associatedNodes), expected nodes: \(reflecting: exactlyNodes),
+                          diff: \(reflecting: diff).
+                        """)
                 }
             }
         }

